wgpu/src/lib.rs

```diff
@@ -887,7 +887,28 @@ static_assertions::assert_impl_all!(RenderBundleDescriptor: Send, Sync);
 ///
 /// Corresponds to [WebGPU `GPUTextureDescriptor`](
 /// https://gpuweb.github.io/gpuweb/#dictdef-gputexturedescriptor).
-pub type TextureDescriptor<'a> = wgt::TextureDescriptor<Label<'a>>;
+#[derive(Clone, Debug)]
+pub struct TextureDescriptor<'a> {
+    /// Debug label of the texture. This will show up in graphics debuggers for easy identification.
+    pub label: Label<'a>,
+    /// Size of the texture. All components must be greater than zero. For a
+    /// regular 1D/2D texture, the unused sizes will be 1. For 2DArray textures,
+    /// Z is the number of 2D textures in that array.
+    pub size: Extent3d,
+    /// Mip count of texture. For a texture with no extra mips, this must be 1.
+    pub mip_level_count: u32,
+    /// Sample count of texture. If this is not 1, texture must have [`BindingType::Texture::multisampled`] set to true.
+    pub sample_count: u32,
+    /// Dimensions of the texture.
+    pub dimension: TextureDimension,
+    /// Format of the texture.
+    pub format: TextureFormat,
+    /// Allowed usages of the texture. If used in other ways, the operation will panic.
+    pub usage: TextureUsages,
+    /// Specifies what view format values will be allowed when calling create_view() on this texture.
+    /// Note: currenly, only srgb-ness is allowed.
+    pub view_formats: &'a [TextureFormat],
```

Comment:
Instead of completely redefining this - can we use the generic trick like we're already using for labels? That was wgc could deal Vec (or Cow) while the surface api can deal with &[]

Reply:
I'm sorry I still haven't figured out how to do this. I looked at structes like `RenderBundleEncoderDescriptor` `RenderPassDescriptor` `VertexBufferLayout` etc that use the `Cow<'a [AType]>` form and they are all redefined in wgpu/lib.rs. And the `Label` generic would make passing `XXDescriptor` always require using `map_label<K>(&self, fun: impl FnOnce(&L) -> K)-> XXDescriptor<K>`, view_formats also defined as a generic would complicate the code.